pyradiosky/skymodel.py

```diff
@@ -1127,6 +1166,277 @@ def point_to_healpix(
+            self.coherency_radec = new_coherency / astropy_healpix.nside_to_pixel_area(
+                self.nside
+            )
+        else:
+            self.hpx_inds = hpx_inds
+            self.stokes = self.stokes / astropy_healpix.nside_to_pixel_area(self.nside)
+            self.coherency_radec = (
+                self.coherency_radec / astropy_healpix.nside_to_pixel_area(self.nside)
+            )
+        self.name = None
+        self.ra = None
+        self.dec = None
+
+        if full_sky and self.Ncomponents < hpx_obj.npix:
+            # add in zero flux pixels
+            new_inds = np.array(
+                set(np.arange(hpx_obj.npix)).difference(set(self.hpx_inds))
+            )
+            n_new = new_inds.size
+            if self.stokes_error is not None:
+                new_stokes_error = np.zeros(
+                    (4, self.Nfreqs, n_new), dtype=self.stokes.dtype
+                )
+            else:
+                new_stokes_error = None
+            if self.reference_frequency is not None:
+                new_reference_frequency = np.full(
+                    n_new, np.median(self.reference_frequency)
+                )
+            else:
+                new_reference_frequency = None
+            if self.spectral_index is not None:
+                new_spectral_index = np.full(n_new, np.median(self.spectral_index))
+            else:
+                new_spectral_index = None
+            if self.beam_amp is not None:
+                new_beam_amp = np.zeros((4, self.Nfreqs, n_new), dtype=self.beam_amp)
+            else:
+                new_beam_amp = None
+            if self.extended_model_group is not None:
+                new_extmod = np.full(n_new, "")
+            else:
+                new_extmod = None
+
+            new_obj = SkyModel(
+                component_type="healpix",
+                nside=self.nside,
+                order=self.order,
+                spectral_type=self.spectral_type,
+                freq_array=self.freq_array,
+                hpx_inds=new_inds,
+                stokes=np.zeros((4, self.Nfreqs, n_new), dtype=self.stokes.dtype),
+                stokes_error=new_stokes_error,
+                reference_frequency=new_reference_frequency,
+                spectral_index=new_spectral_index,
+                beam_amp=new_beam_amp,
+                extended_model_group=new_extmod,
+            )
+            self.concat(new_obj)
+
+        # sort in order of hpx_inds:
+        sort_order = np.argsort(self.hpx_inds)
+        self.hpx_inds = self.hpx_inds[sort_order]
+        self.stokes = self.stokes[:, :, sort_order]
+        self.coherency_radec = self.coherency_radec[:, :, :, sort_order]
+        if self.stokes_error is not None:
+            self.stokes_error = self.stokes_error[:, :, sort_order]
+        for param in self.ncomponent_length_params:
+            attr = getattr(self, param)
+            param_name = attr.name
```

Comment:
shouldn't `param_name` be equivalent to `param`?

Reply:
No. The `param` in this case is a UVParameter object (e.g. `_spectral_index`) while the `param_name` is the property name on the SkyModel object (e.g. `spectral_index`).

Reply:
Ah.